samplers/samplers_test.go

```diff
@@ -76,6 +76,58 @@ func TestCounterMerge(t *testing.T) {
 	assert.Equal(t, float64(3.8), metrics[0].Value[0][1])
 }
 
+func TestCardinality(t *testing.T) {
+	rand.Seed(time.Now().Unix())
+	c := NewCardinalityCount()
+	c.Sample("abc", "tag1tag2tag3")
+
+	for i := 0; i < 100; i++ {
+		c.Sample("abc", strconv.Itoa(rand.Int()))
+	}
+	assert.Equal(t, uint64(101), c.MetricCardinality["abc"].Count(), "counts did not match")
+}
+
+func TestCardinalityMerge(t *testing.T) {
+	rand.Seed(time.Now().Unix())
+
+	cc := NewCardinalityCount()
+	for i := 0; i < 100; i++ {
+		cc.Sample("abc", strconv.Itoa(rand.Int()))
+	}
+	for i := 0; i < 150; i++ {
+		cc.Sample("cde", strconv.Itoa(rand.Int()))
+	}
+	assert.Equal(t, uint64(100), cc.MetricCardinality["abc"].Count(), "counts did not match")
+	assert.Equal(t, uint64(150), cc.MetricCardinality["cde"].Count(), "counts did not match")
+
+	jsonMetrics, err := cc.ExportSets()
+	assert.NoError(t, err, "should have exported successfully")
+	assert.NotEmpty(t, jsonMetrics, "should have returned a value")
+	assert.NotNil(t, jsonMetrics[0], "Need a value")
+
+	cc2 := NewCardinalityCount()
+	for i := 0; i < 50; i++ {
+		cc2.Sample("abc", strconv.Itoa(rand.Int()))
+	}
+	for i := 0; i < 50; i++ {
+		cc2.Sample("cde", strconv.Itoa(rand.Int()))
+	}
+
+	for _, jm := range jsonMetrics {
+		assert.NoError(t, cc2.Combine(jm.Value), "should have combined successfully")
+	}
+
+	// // HLLs are approximate, and we've seen error of +-1 here in the past, so
+	// // we're giving the test some room for error to reduce flakes
```

Comment:
Can we extract the margin of error into a constant?

Reply:
You can use [InDelta](https://godoc.org/github.com/stretchr/testify/assert#InDelta) here!